pkg/models/terminal/terminal.go

```diff
@@ -378,21 +379,33 @@ func (t *terminaler) HandleShellAccessToNode(nodename string, conn *websocket.Co
 }
 
 func (n *NodeTerminaler) WatchPodStatusBeRunning(pod *v1.Pod, succ chan bool, fail chan bool) {
-	var err error
-	for i := 0; i < 5; i++ {
-		if pod.Status.Phase == v1.PodRunning {
-			idx, _ := NodeSessionCounter.Load(n.Nodename)
-			atomic.AddInt64(idx.(*int64), 1)
-			close(succ)
-			return
-		}
-		time.Sleep(time.Second)
+	if pod.Status.Phase == v1.PodRunning {
+		idx, _ := NodeSessionCounter.Load(n.Nodename)
+		atomic.AddInt64(idx.(*int64), 1)
+		close(succ)
+		return
+	}
+
+	err := wait.Poll(time.Millisecond*500, time.Second*5, func() (done bool, err error) {
 		pod, err = n.client.CoreV1().Pods(pod.ObjectMeta.Namespace).Get(context.Background(), pod.ObjectMeta.Name, metav1.GetOptions{})
 		if err != nil {
 			klog.Warning(err)
-			close(fail)
-			return
+			return false, err
+		}
+
+		if pod.Status.Phase == v1.PodRunning {
+			idx, _ := NodeSessionCounter.Load(n.Nodename)
+			atomic.AddInt64(idx.(*int64), 1)
+			return true, nil
 		}
+
+		return false, nil
+	})
+
+	if err != nil {
+		klog.Warning("watching pod status error: ", err)
+		close(fail)
```

Comment:
Last time, I recommend using two channels and one goroutine to select signal in order to consider adding `time.After` to do timeout logic to cancel the context,  so far I think these two channels and goroutine can be removed, we can directly call the following logic here, the codes would be more clear and simple:
```
t.HandleSession(nodeTerminaler.Shell, nodeTerminaler.Namespace, nodeTerminaler.PodName, nodeTerminaler.ContainerName, conn)
		defer nodeTerminaler.CleanUpNSEnterPod()
```
What do you think? @lynxcat @zryfish